app/code/Magento/Inventory/Indexer/Scope/UnknownStateException.php

```diff
@@ -0,0 +1,18 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Indexer\Scope;
+
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * Exception for situation where used state which is not defined in configuration
+ * @todo refactoring it copy from catalog search module
+ * @api
+ */
+class UnknownStateException extends LocalizedException
```

Comment:
@naydav agree, already pointed that out for IndexTableNotExistException.
These should be framework level exceptions, we can't provide them to client because they provide details of how this functionality is implemented (leaky abstractions).

So, these exceptions are not successors of Localized